File: Remove_Nth_Node_From_End_of_List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int sizz(ListNode *head){
        int n = 0;
        while(head != NULL){
            n ++;
            head = head->next;
        }
        return n;
    }
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        int ind = sizz(head) - n;
        if(ind ==0){
            head = head->next;
            return head;
        }
        else if(ind == sizz(head)-1){
            ListNode * temp = head;
            ListNode * prev = NULL;

            while(temp->next != NULL){
                prev = temp;
                temp = temp->next;
            }
            prev->next= NULL;
            temp = NULL;
            return head;

        }
        else{
            ListNode * temp = head;
            int ii= 0;
            ListNode * prev = NULL;
            while(temp != NULL){
                if(ii == ind){
                    ListNode * dele = temp;
                    prev->next = temp->next;
                    delete dele;
                    return head;

                }
                prev = temp;
                temp = temp->next;
                ii ++;
            }
        }
        return head;
        
    }
};

File: Remove_Nth_Node_From_End_of_List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        temp = head
        arr = []
        while(temp != None):
            arr.append(temp.val)
            temp = temp.next
        nn = len(arr)
        arr.pop(nn - n)
        head = None
        tail = None
        for i in arr:
            temp = ListNode(i)
            if(head == None):
                head = temp
                tail = temp
            else:
                tail.next = temp
                tail = temp
        return head

        
